lib/libc/picolibc/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+zephyr_library_sources(libc-hooks.c)
+
+# Zephyr normally uses -ffreestanding, which with current GNU toolchains
+# means that the flag macros used by picolibc <inttypes.h> to signal
+# support for PRI.64 macros are not present.  To make them available we
+# need to hook into the include path before the system files and
+# explicitly include the picolibc header that provides those macros.
+zephyr_include_directories(include)
+
+# define __LINUX_ERRNO_EXTENSIONS__ so we get errno defines like -ESHUTDOWN
+# used by the network stack
+zephyr_compile_definitions(__LINUX_ERRNO_EXTENSIONS__)
+
+zephyr_link_libraries(
+  m
+  c
+  gcc # Lib C depends on libgcc.
```

Comment:
No changes requested, but with the fact that Zephyr is working towards more toolchains, especially LLVM, and thus  also the support of compiler-rt runtime library, I was wondering if picolib could be used in that case.

Also this makes me wonder if the use of picolib should be restricted, based on toolchain in use.
But that would probably need to go into a bigger work at later time.

Reply:
picolibc built with llvm still requires libgcc as that has all of the helper functions required by the underlying ABI, such as the __aeabi functions for ARM. -lgcc would be automatically inserted into the link command by the compiler front-end (either clang or gcc) if zephyr wasn't using -nostdlib. zephyr could probably remove that flag and simply let the compiler front end insert both -lc and -lgcc.